### PR TITLE
Adding unit tests to cover nil pointer cases of Include()

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

### CODE REVIEW

File: pkg/manifest/manifest_test.go [hunk 1]
```diff
@@ -587,39 +587,62 @@ func setupTestFS(t *testing.T, d dir) (string, func() error) {
...
 		},
 		{
 			name:        "profile selection works",
-			profile:     "single-node",
+			profile:     &singleNodeProfile,
 			annotations: map[string]interface{}{"include.release.openshift.io/self-managed-high-availability": "true"},
 			expected:    fmt.Errorf("include.release.openshift.io/single-node unset"),
 		},
+		{
+			name:        "No profile",
```
Comment (jottofar) [inline pkg/manifest/manifest_test.go:632]: Change this to "all nil", which I think it is, and cover all nil's with one case. Or add a case for each one set to nil.

File: pkg/manifest/manifest_test.go [hunk 2]
```diff
@@ -587,39 +587,62 @@ func setupTestFS(t *testing.T, d dir) (string, func() error) {
...
-		profile            string
+		exclude            *string
+		includeTechPreview *bool
+		profile            *string
 		annotations        map[string]interface{}
-		caps               configv1.ClusterVersionCapabilitiesStatus
+		caps               *configv1.ClusterVersionCapabilitiesStatus
 
 		expected error
 	}{
```
Comment (jottofar) [inline pkg/manifest/manifest_test.go:605]: add "no annotations" test

File: pkg/manifest/manifest_test.go [hunk 3]
```diff
@@ -656,42 +680,42 @@ func Test_include(t *testing.T) {
...
 			annotations: map[string]interface{}{},
 			expected:    fmt.Errorf("include.release.openshift.io/self-managed-high-availability unset"),
 		},
 		{
 			name:        "default profile selection excludes with no annotation",
-			profile:     DefaultClusterProfile,
+			profile:     &defaultClusterProfile,
 			annotations: nil,
 			expected:    fmt.Errorf("no annotations"),
 		},
```
Comment (jottofar) [inline pkg/manifest/manifest_test.go:692]: add "no capability anno" test

File: pkg/manifest/manifest_test.go (hunk 1 above)
Comment (LalatenduMohanty) [reply pkg/manifest/manifest_test.go:632]: I will add an all nil use case which will be equivalent to no annotation test case. The include method reacts differently when each of the input is nil so I have handled each use case differently.

File: pkg/manifest/manifest_test.go (hunk 3 above)
Comment (LalatenduMohanty) [reply pkg/manifest/manifest_test.go:692]: Now that capability is pointer i.e. `caps    *configv1.ClusterVersionCapabilitiesStatus` , when ever we not passing any capability it is passed as `nil`.

File: pkg/manifest/manifest_test.go (hunk 2 above)
Comment (LalatenduMohanty) [reply pkg/manifest/manifest_test.go:605]: Fixed https://github.com/openshift/library-go/commit/7ae641e6db34e5fad1456c5954b03f2b51292cae

File: pkg/manifest/manifest_test.go (hunk 1 above)
Comment (LalatenduMohanty) [reply pkg/manifest/manifest_test.go:632]: Fixed in https://github.com/openshift/library-go/commit/7ae641e6db34e5fad1456c5954b03f2b51292cae

File: pkg/manifest/manifest_test.go (hunk 3 above)
Comment (jottofar) [reply pkg/manifest/manifest_test.go:692]: Right, but I was referring to something more explicit but it's not needed.
